pkg/healthcheck/healthcheck_multicluster.go

```diff
@@ -54,108 +59,207 @@ var expectedServiceMirrorRemoteClusterPolicyVerbs = []string{
+				{
+					description: "all target cluster gateways are alive",
+					hintAnchor:  "l5d-multicluster-target-gateways-alive",
+					check: func(ctx context.Context) error {
+						return hc.checkRemoteClusterGatewaysHealth(ctx)
+					},
 				},
-			},
-			{
-				description: "clusters share trust anchors",
-				hintAnchor:  "l5d-multicluster-clusters-share-anchors",
-				check: func(ctx context.Context) error {
-					return hc.checkRemoteClusterAnchors()
+				{
+					description: "clusters share trust anchors",
+					hintAnchor:  "l5d-multicluster-clusters-share-anchors",
+					check: func(ctx context.Context) error {
+						return hc.checkRemoteClusterAnchors()
+					},
+				},
+				{
+					description: "multicluster daisy chaining is avoided",
+					hintAnchor:  "l5d-multicluster-daisy-chaining",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkDaisyChains()
+					},
+				},
+				{
+					description: "all mirror services have endpoints",
+					hintAnchor:  "l5d-multicluster-services-endpoints",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkIfMirroredServicesHaveEndpoints()
+					},
+				},
+				{
+					description: "all gateway mirrors have endpoints",
+					hintAnchor:  "l5d-multicluster-gateways-endpoints",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkIfGatewayMirrorsHaveEndpoints()
+					},
+				},
+				{
+					description: "all mirror services have a valid gateway mirror",
+					hintAnchor:  "l5d-multicluster-mirror-gateways",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkIfAllMirrorServicesHaveGatewayMirrors()
+					},
+				},
+				{
+					description: "remote: all gateways have external IPs",
+					hintAnchor:  "l5d-multicluster-gateways-exist",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkRemoteIfGatewaysHaveExternalIP()
+					},
+				},
+				{
+					description: "remote: gateways exists for all exported services",
+					hintAnchor:  "l5d-multicluster-exported-gateways",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkRemoteifGatewaysExistforExportedServices()
+					},
+				},
+				{
+					description: "remote: all gateways have correct port names exposed",
+					hintAnchor:  "l5d-multicluster-gateways-ports",
+					warning:     true,
+					check: func(ctx context.Context) error {
+						return hc.checkRemoteIfAllGatewaysHaveCorrectPortNames()
+					},
 				},
 			},
-			{
-				description: "multicluster daisy chaining is avoided",
-				hintAnchor:  "l5d-multicluster-daisy-chaining",
-				warning:     true,
-				check: func(ctx context.Context) error {
-					return hc.checkDaisyChains()
+		},
+		{
+			id: LinkerdMulticlusterTargetChecks,
+			checkers: []checker{
+				{
+					description: "all cluster gateways are alive",
```

Comment:
My same suggestion about combining the gateway checks applies here as well.  We can do a single check that verifies that each exported service has a valid gateway (where "valid" means that it exists, has an external IP, has the right ports, has non-empty endpoints)